llvm/include/llvm/ProfileData/IndexedMemProfData.h

```diff
@@ -89,7 +90,7 @@ struct IndexedMemProfData {
 Error writeMemProf(
     ProfOStream &OS, memprof::IndexedMemProfData &MemProfData,
     memprof::IndexedVersion MemProfVersionRequested, bool MemProfFullSchema,
-    std::unique_ptr<memprof::DataAccessProfData> DataAccessProfileData);
-
+    std::unique_ptr<memprof::DataAccessProfData> DataAccessProfileData,
+    memprof::MemProfSummary *MemProfSum);
```

Comment:
There's no need to keep the memprof summary around after we call `writeMemProf`, since it's a unique_ptr already can we just move it here instead of passing a pointer?